### PR TITLE
DRAFT: Gitea drone

### PR DESCRIPTION
**Pull request by Morriz**

### CODE REVIEW

File: bin/gen-drone.sh [hunk 1]
```diff
@@ -26,8 +26,15 @@ fi
 readonly webhook=$(echo "$values" | yq r - "alerts.$receiver.$key")
 
 function template_drone_config() {
+<<<<<<< HEAD
```
Comment (j-zimnowoda) [inline bin/gen-drone.sh:29]: please solve merge conflicts

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]

File: bin/gitea-push.sh [hunk 2]
```diff
@@ -13,25 +13,34 @@ readonly gitea_url="gitea.$cluster_domain"
...
-else
+  read "Another origin already exists, do you want to add Gitea as a remote? [y/N]" add_remote
+  if [ $add_remote == 'y' ] || [ $add_remote == 'Y' ]; then
+    remote_name='otomi-values'
+  else
+    exit 0
+  fi
+fi
+if [ ! $(git config remote.$remote_name.url) ]; then
   git remote add $remote_name "https://$gitea_user:$gitea_password@$gitea_url/$gitea_user/$gitea_repo.git"
```
Comment (svatwork) [inline bin/gitea-push.sh:37]: This assumes HTTPS, no? Do we assume a certificate (self-signed or not) is available?

File: bin/gitea-push.sh [hunk 3]
```diff
@@ -13,25 +13,34 @@ readonly gitea_url="gitea.$cluster_domain"
...
   echo "Added gitea as a remote origin"
+  echo "You can push using: \`git push main $remote_name\`"
 fi
 
 # Try to pull, if repo is not new, it will get data
-git fetch $remote_name master || true
+git fetch $remote_name main || true
 # Which will show how many commits are there.
 readonly commit_count=$(git rev-list --count --remotes=$remote_name)
```
Comment (svatwork) [inline bin/gitea-push.sh:45]: Just curious if it's even possible to pull a remote with <= 0 commits? 

I guess you want to check if there are any existing repos we need to take into consideration? If not, why not just force push (assuming force push is idempotent)?

File: values/drone/drone.gotmpl [hunk 4]
```diff
@@ -39,11 +41,17 @@ server:
...
-    DRONE_REPOSITORY_FILTER: {{ . }}
     {{- end }}
+<<<<<<< HEAD
+    {{- if hasKey $d "repoFilter" }}
+    DRONE_REPOSITORY_FILTER: {{ $d.repoFilter }}
+    {{- end }}
+    DRONE_SERVER_PROXY_HOST: {{ $host }}/hook
+=======
     DRONE_SERVER_PROXY_HOST: proxy.{{ $v.dns.domain }}/drone
+>>>>>>> master
```
Comment (svatwork) [inline values/drone/drone.gotmpl:54]: Another merge conflict here

File: charts/gitea/templates/gitea/init.yaml [hunk 5]
```diff
@@ -16,6 +16,8 @@ stringData:
     # END: initPreScript
     {{- end }}
 
+    chmod 777 /data
```
Comment (svatwork) [inline charts/gitea/templates/gitea/init.yaml:19]: How come chmod 777 is necessary? Sounds like a bad idea?

File: bin/deploy.sh [hunk 6]
```diff
@@ -8,7 +8,10 @@ hf -f helmfile.tpl/helmfile-init.yaml template | kubectl apply -f -
 kubectl apply -f charts/prometheus-operator/crds
 
 # helm charts after
-hf apply --skip-deps
+hf apply --skip-deps -l stage!=post
```
Comment (j-zimnowoda) [inline bin/deploy.sh:11]: I see a potential problem with `stage!=post`
E.g.:
```
otomi apply -f helmfile.d/helmfile-30.admin.yaml
```
will deploy every release from `helmfile-30.admin.yaml` 

Do we accept it?

File: bin/retrieve-credentials.sh [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+prepare_crypt
+readonly values=$(hf_values)
+readonly gitea_enabled=$(echo "$values" | yq r - 'charts.gitea.enabled')
+readonly drone_enabled=$(echo "$values" | yq r - 'charts.drone.enabled')
+
+[ "$gitea_enabled" != "true" ] && echo "Gitea is disabled" && exit 0
+[ "$drone_enabled" != "true" ] && echo "Drone is disabled" && exit 0
+
+readonly domain=$(echo "$values" | yq r - 'cluster.domain')
```
Comment (j-zimnowoda) [inline bin/retrieve-credentials.sh:17]: `yq r - 'dns.domain'`

File: bin/deploy.sh (hunk 6 above)
Comment (Morriz) [reply bin/deploy.sh:11]: That will deploy everything except the ones with `stage=post` which is what we want to yes.

File: charts/gitea/templates/gitea/init.yaml (hunk 5 above)
Comment (Morriz) [reply charts/gitea/templates/gitea/init.yaml:19]: I deem this a gitea bug for now, 777 is fine until that fix comes back

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
LGTM
Please take a look at my few comments.

**Review by Morriz — COMMENTED** [review]
please merge with master

**Pull request merged**